tests/test_downloadermiddleware_retry.py

```diff
@@ -91,12 +93,12 @@ def test_twistederrors(self):
             ConnectError,
             ConnectionDone,
             ConnectionLost,
-            ConnectionRefusedError,
+            TxConnectionRefusedError,
             defer.TimeoutError,
             DNSLookupError,
             ResponseFailed,
             TCPTimedOutError,
-            TimeoutError,
+            TxTimeoutError,
```

Comment:
This checked builtin exceptions instead of Twisted ones, the test was passing because the builtin exceptions are subclasses of OSError which is retried.

Reply:
Great catch!